example/auth/main.go

```diff
@@ -0,0 +1,57 @@
+package main
```

Comment:
I think we should provide a more practical and simpler example of a usecase where a client wants to do several concurrent `ondemand`, `structured-contents`, etc. requests  - in all those concurrent calls, they will rely on the helper API to keep on providing a valid access token.

Something like this.....
```
func main() {
	// Initialize the authentication client and helper
	// We rely on authentication helper to return a valid access token for all our requests. The helper APIs will internally refresh, re-login as necessary.
	authClient := auth.NewClient()
	helper, err := auth.NewHelper(authClient)
	if err != nil {
		log.Fatalln(err)
	}

	
	var wg sync.WaitGroup
	ctx := context.Background()
	reqs := 50
	wg.Add(reqs)

	for i := 0; i < reqs; i++ {
		go func(req int) {
			defer wg.Done()
			// Get access token
			tkn, err := helper.GetAccessToken()
			if err != nil {
				log.Printf("failed to get access token for request %d: %v", req, err)
				return
			}

			// Call any WME API
			clt := api.NewClient()
			clt.SetAccessToken(tkn)
			arq := &api.Request{
				Fields: []string{"name", "abstract", "url", "version"},
				Filters: []*api.Filter{
					{
						Field: "in_language.identifier",
						Value: "en",
					},
				},
			}
			
			scs, err := clt.GetArticles(ctx, "Montreal", req)
			if err != nil {
				log.Printf("failed to get articles for request %d: %v", req, err)
				return
			}
		}(i)
	}

	
	wg.Wait()

	// Revoke token and remove state from storage when we are done using WME APIs
	if err := helper.ClearState(); err != nil {
		log.Printf("failed to clear state: %v", err)
	}
}

Reply:
Fixed